src/drivers/ins/eulernav_bahrs/eulernav_driver.cpp

```diff
@@ -0,0 +1,526 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2025 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "eulernav_driver.h"
+#include <px4_platform_common/getopt.h>
+#include <drivers/drv_sensor.h>
+#include <matrix/Quaternion.hpp>
+#include <matrix/Euler.hpp>
+#include <atmosphere/atmosphere.h>
+
+EulerNavDriver::EulerNavDriver(const char *device_name)
+	: ModuleParams{nullptr}
+	, _serial_port{device_name, 115200, ByteSize::EightBits, Parity::None, StopBits::One, FlowControl::Disabled}
+	, _data_buffer{}
+	, _px4_accel{DRV_INS_DEVTYPE_BAHRS}
+	, _px4_gyro{DRV_INS_DEVTYPE_BAHRS}
+	, _attitude_pub{ORB_ID(vehicle_attitude)}
+	, _barometer_pub{ORB_ID(sensor_baro)}
+{
+	initialize();
```

Comment:
don't initialize in the constructor. Instead init in the `instantiate` function. This way if something fails your driver can exit gracefully. For example if there is not enough memory to allocate.